warehouse/email/__init__.py

```diff
@@ -33,12 +35,25 @@ def _compute_recipient(user, email):
 
 
 @tasks.task(bind=True, ignore_result=True, acks_late=True)
-def send_email(task, request, recipient, msg):
+def send_email(task, request, recipient, msg, user_id, ip_address):
```

Comment:
The `send_email` task is also used in the admin: https://github.com/pypa/warehouse/blob/de634ab4f46adf34cb5bb1af701f1f807df01343/warehouse/admin/views/emails.py#L95-L102

In that use case, the IP address should always be redacted, so it might make more sense to pass `redact_ip` in instead.

Also we should be more clear what `user_id` is here. Something like:

```suggestion
def send_email(task, request, recipient, msg, sending_user_id, redact_ip):
```